### ISSUE TITLE
How can I zoom/scroll within the TimeSamplePlot ?

### ISSUE THREAD

**Issue by anox1337** [KoalaPlot/koalaplot-core#69]
Hello!

I tried out the KoalaPlot Time Chart example and it works like a charm. But I cannot get it to zoom / scroll.
Like it is not even registering the gestures.

`    override fun zoom(zoomFactor: Float, pivot: Float) {
`
is not being called.

I tried comparing the code with the LogLineChartSample but really cannot figure out what I doing wrong. Could somebody give me a pointer? :) 

```
    ChartLayout(
            modifier = paddingMod,
            title = { ChartTitle(title) },
            legendLocation = LegendLocation.BOTTOM
        ) {
            XYGraph(
                xAxisModel = LongLinearAxisModel(
                    allowZooming = true,
                    allowPanning = true,
                    range = (data.first().x)..(data.last().x) + 1, // <-- epochSeconds from 01.01.2024 - 31.12.2024
                ),
                yAxisModel = IntLinearAxisModel(
                    allowPanning = true,
                    allowZooming = true,
                    range = yDataMin..yDataMax // <-- 0 .. 33000
                ),
                xAxisLabels = {
                    if (!thumbnail) {
                        AxisLabel(Instant.fromEpochSeconds(it).toString(), Modifier.padding(top = 2.dp))
                    }
                },
                xAxisStyle = rememberAxisStyle(labelRotation = 90),
                xAxisTitle = {
                    if (!thumbnail) {
                        Box(
                            contentAlignment = Alignment.Center
                        ) {
                            AxisTitle("Time")
                        }
                    }
                },
                yAxisLabels = {
                    if (!thumbnail) AxisLabel(it.toString(), Modifier.absolutePadding(right = 2.dp))
                },
                yAxisTitle = {
                    if (!thumbnail) {
                        Box(
                            modifier = Modifier.fillMaxHeight(),
                            contentAlignment = Alignment.TopStart
                        ) {
                            AxisTitle(
                                "Value",
                                modifier = Modifier.rotateVertically(VerticalRotation.COUNTER_CLOCKWISE)
                                    .padding(bottom = padding)
                            )
                        }
                    }
                }
            ) {
                chart(data) // <-- DataPoints are List of: DefaultPoint<Long, Int>
            }
```

**Comment by gsteckman** [comment]
Zooming only works on Android because, last I checked, the Compose multiplatform mouse gesture handling doesn't support the same features. Have you tried on Android?

**Comment by gsteckman** [comment]
I tried on the sample and it's not working. When debugging it, it looks like the original xAxisModel from when the XYGraph was first created is being held onto. Since that axis has a range of 1 it's not possible to zoom it. I can't figure out why when more data points are added the pointer input modifier isn't being invoked with the new xAxisModel. It's going to take more troubleshooting to track down what's going on here.

I suspect this same behavior is present in the other samples as well but in those the data set is constant so the axis models never change from the original one used to create the XYGraph the first time.

**Comment by gsteckman** [comment]
I think I found it, I didn't set key values when creating the pointer input modifier in XYGraph, so it's not getting restarted when the axis models change. I won't be able to get to trying the fix for a few days to a week.

**Comment by gsteckman** [comment]
This is now fixed in 0.6.2.
